### PR TITLE
DFR-3468 Check applicant represented before adding app sol policy

### PR DESCRIPTION
**Pull request by al-hmcts**
### Jira link

See [DFR-3468](https://tools.hmcts.net/jira/browse/DFR-3468)

### Change description
- Check Applicant represented before adding app sol policy
- update unit tests

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateConsentedSubmittedHandlerTest.java [hunk 1]
```diff
@@ -37,38 +41,73 @@ public class SolicitorCreateConsentedSubmittedHandlerTest {
...
+            .when(assignApplicantSolicitorService)
+            .setApplicantSolicitor(any(), any());
+
+        GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> response = handler.handle(callbackRequest, AUTH_TOKEN);
+
+        verify(createCaseService, times(1)).setSupplementaryData(callbackRequest, AUTH_TOKEN);
+        verify(assignApplicantSolicitorService, times(1))
+            .setApplicantSolicitor(callbackRequest.getCaseDetails(), AUTH_TOKEN);
+
+        assertTrue(response.getErrors().contains("Exception message"));
```
Comment (hamzahtahirhmcts) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateConsentedSubmittedHandlerTest.java:90]: should we have `assertEquals(1, response.getErrors().size());` too?

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateConsentedSubmittedHandlerTest.java [hunk 2]
```diff
@@ -37,38 +41,73 @@ public class SolicitorCreateConsentedSubmittedHandlerTest {
...
+        FinremCaseData caseData = FinremCaseData.builder()
+            .contactDetailsWrapper(ContactDetailsWrapper.builder().applicantRepresented(isApplicantRepresented).build())
+            .build();
         FinremCaseDetails caseDetails = FinremCaseDetails.builder().id(123L).data(caseData).build();
         return FinremCallbackRequest.builder().eventType(EventType.SOLICITOR_CREATE).caseDetails(caseDetails).build();
     }
-
-}
\ No newline at end of file
+}
```
Comment (hamzahtahirhmcts) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateConsentedSubmittedHandlerTest.java:113]: NL

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateContestedSubmittedHandlerTest.java [hunk 3]
```diff
@@ -38,38 +42,72 @@ public class SolicitorCreateContestedSubmittedHandlerTest {
...
+        FinremCaseData caseData = FinremCaseData.builder()
+            .contactDetailsWrapper(ContactDetailsWrapper.builder().applicantRepresented(isApplicantRepresented).build())
+            .build();
         FinremCaseDetails caseDetails = FinremCaseDetails.builder().id(123L).data(caseData).build();
         return FinremCallbackRequest.builder().eventType(EventType.SOLICITOR_CREATE).caseDetails(caseDetails).build();
     }
-
-}
\ No newline at end of file
+}
```
Comment (hamzahtahirhmcts) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateContestedSubmittedHandlerTest.java:113]: NL

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateContestedSubmittedHandlerTest.java [hunk 4]
```diff
@@ -38,38 +42,72 @@ public class SolicitorCreateContestedSubmittedHandlerTest {
...
+            .setApplicantSolicitor(callbackRequest.getCaseDetails(), AUTH_TOKEN);
+
+        assertTrue(response.getErrors().contains("Exception message"));
+        assertTrue(response.getErrors()
+            .contains("Failed to assign applicant solicitor to case, "
+                + "please ensure you have selected the correct applicant organisation on case"));
+    }
+
+    @Test
+    void givenACcdCallbackSolicitorCreateContestedCase_WhenHandleWithNoAppRepresentation_thenApplicantSolicitorNotAssigned() {
```
Comment (hamzahtahirhmcts) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateContestedSubmittedHandlerTest.java:97]: I like the namings on these tests, nice 👍

Comment (al-hmcts) [reply src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateContestedSubmittedHandlerTest.java:97]: Thanks :)

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateConsentedSubmittedHandlerTest.java (hunk 2 above)
Comment (al-hmcts) [reply src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateConsentedSubmittedHandlerTest.java:113]: New line is there, I think your seeing where I fixed the EOL in the file prior to my change.

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateContestedSubmittedHandlerTest.java (hunk 3 above)
Comment (al-hmcts) [reply src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateContestedSubmittedHandlerTest.java:113]: New line is there, I think your seeing where I fixed the EOL in the file prior to my change.

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateConsentedSubmittedHandlerTest.java (hunk 1 above)
Comment (al-hmcts) [reply src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateConsentedSubmittedHandlerTest.java:90]: Yeah not a bad shout

**Review by hamzahtahirhmcts — APPROVED** [review]

**Review by jthmcts — APPROVED** [review]

**Pull request merged**